internal/source/git.go

```diff
@@ -231,10 +237,26 @@ func (f *billyFS) ReadDir(name string) ([]fs.DirEntry, error) {
 
 type billyFile struct {
 	billy.File
-	fi    os.FileInfo
-	fiErr error
+	fi os.FileInfo
 }
 
 func (b billyFile) Stat() (fs.FileInfo, error) {
-	return b.fi, b.fiErr
+	return b.fi, nil
+}
+
+func (b billyFile) Close() error {
+	if b.File != nil {
+		return b.File.Close()
+	}
+	return nil
+}
```

Comment:
Should we also implement a similar custom `Read` function on `billyFile` to avoid a panic when `b.File` is `nil` (in the case of a billyDirFile)?